core/utils.c

```diff
@@ -4649,10 +4649,7 @@ stats_get_snapshot(dr_stats_t *drstats)
     if (!GLOBAL_STATS_ON())
         return false;
     CLIENT_ASSERT(drstats != NULL, "Expected non-null value for parameter drstats.");
-    /* We are at V1 of the structure, and we can't return less than the one
-     * field. We need to remove this assert when we add more fields.
-     */
-    CLIENT_ASSERT(drstats->size >= sizeof(dr_stats_t), "Invalid drstats->size value.");
     drstats->basic_block_count = GLOBAL_STAT(num_bbs);
+    drstats->peak_num_threads = GLOBAL_STAT(peak_num_threads);
```

Comment:
This is ignoring compatibility: it needs to check the size to ensure this field exists in the client.

Reply:
Done.